dmodel/dm-macros.h

```diff
@@ -0,0 +1,144 @@
+/* Copyright 2018  Endless Mobile, Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <glib.h>
+#include "dm-version.h"
+
+#ifndef _DM_EXTERN
+#define _DM_EXTERN extern
+#endif
+
+#ifdef DM_DISABLE_DEPRECATION_WARNINGS
+#define DM_DEPRECATED _DM_EXTERN
+#define DM_DEPRECATED_FOR(f) _DM_EXTERN
+#define DM_UNAVAILABLE(maj,min) _DM_EXTERN
+#else
+#define DM_DEPRECATED G_DEPRECATED _DM_EXTERN
+#define DM_DEPRECATED_FOR(f) G_DEPRECATED_FOR(f) _DM_EXTERN
+#define DM_UNAVAILABLE(maj,min) G_UNAVAILABLE(maj,min) _DM_EXTERN
+#endif
+
+/* Each new cycle should add a new version symbol here */
+
+/**
+ * DM_VERSION_0_0:
+ *
+ * A pre-processor macro that evaluates to the 0.0 version of DModel, in a
+ * format that can be used by %DM_VERSION_MIN_REQUIRED and
+ * %DM_VERSION_MAX_ALLOWED.
+ */
+#define DM_VERSION_0_0 (G_ENCODE_VERSION (0, 0))
```

Comment:
I'd honestly drop this: it's kind of pointless to tag at 0.0.

Reply:
I can imagine it being useful for defining `DM_VERSION_MAX_ALLOWED` to 0.0. I removed some of the other stuff, but for now I kept it for that reason.